osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -697,7 +725,10 @@ private void ensureValidStateSwitch(ServerMultiplayerRoom room, MultiplayerUserS
             switch (newState)
             {
                 case MultiplayerUserState.Idle:
-                    // any state can return to idle.
+                    if (isGameplayState(oldState))
+                        throw new InvalidStateException("Cannot abort gameplay in a non-gameplay state.");
```

Comment:
Message reads weird. Maybe something like `Cannot transition out of a gameplay state without aborting.`?